src/buf/buf_mut.rs

```diff
@@ -1419,6 +1419,41 @@ unsafe impl BufMut for &mut [u8] {
     }
 }
 
+unsafe impl BufMut for &mut [core::mem::MaybeUninit<u8>] {
+    #[inline]
+    fn remaining_mut(&self) -> usize {
+        self.len()
+    }
+
+    #[inline]
+    fn chunk_mut(&mut self) -> &mut UninitSlice {
+        UninitSlice::from_slice(self)
+    }
+
+    #[inline]
+    unsafe fn advance_mut(&mut self, cnt: usize) {
+        // Lifetime dance taken from `impl Write for &mut [u8]`.
+        let (_, b) = core::mem::replace(self, &mut []).split_at_mut(cnt);
+        *self = b;
+    }
+
+    #[inline]
+    fn put_slice(&mut self, src: &[u8]) {
+        self.chunk_mut().copy_from_slice(src);
+        unsafe {
+            self.advance_mut(src.len());
+        }
+    }
```

Comment:
This function is wrong. It will panic for `src.len() != self.len()` since `copy_from_slice` has that behavior.

Can you please add some simple tests for these methods? Something along these lines:
```
for len in 0..100 {
    for len2 in 0..=len {
        // Make a `&mut [MaybeUninit<u8>]` of length `len` and call `put_slice` with a slice of `len2`.
    }
}
```
Then also include another test marked should-panic test for `len2 > len`.

Reply:
D’oh.  Done.